File: backend/app/schemas/allocate.py
"""资源分配 API Schema（M2）。"""

from __future__ import annotations

from pydantic import BaseModel, Field


class AllocateRequest(BaseModel):
    """``POST /allocate`` 请求体。"""

    target_type: str = Field(
        ...,
        description="realm=投入境界 | body_temper=投入淬体进度 | technique=升级功法",
    )
    target_id: str | None = None
    amount: int = Field(..., ge=1)

File: frontend/src/types/allocate.ts
/**
 * 资源分配 API 类型（M2）。
 */
import type { CharacterPublic } from './character'

export type AllocateTargetType = 'realm' | 'technique'

export interface AllocateRequest {
  target_type: AllocateTargetType
  target_id?: string | null
  amount: number
}

export interface AllocateResult {
  allocated: number
  levels_gained: number
  message: string
  character: CharacterPublic
}
